airflow/providers/amazon/aws/hooks/redshift_data.py

```diff
@@ -0,0 +1,152 @@
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Interact with AWS Redshift clusters."""
+
+from typing import Optional
+
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+
+
+class RedshiftDataHook(AwsBaseHook):
+    """
+    Interact with AWS Redshift Data, using the boto3 library
+
+    Additional arguments (such as ``aws_conn_id``) may be specified and
+    are passed down to the underlying AwsBaseHook.
+
+    .. seealso::
+        :class:`~airflow.providers.amazon.aws.hooks.base_aws.AwsBaseHook`
+
+    :param aws_conn_id: The Airflow connection used for AWS credentials.
+    :type aws_conn_id: str
+    """
+
+    def __init__(self, *args, **kwargs) -> None:
+        kwargs["client_type"] = "redshift-data"
+        super().__init__(*args, **kwargs)
+
+    def execute_statement(
+        self,
+        cluster_identifier: str,
+        database: str,
+        sql: str,
+        db_user: Optional[str] = "",
+        parameters: Optional[list] = None,
+        secret_arn: Optional[str] = "",
+        statement_name: Optional[str] = "",
+        with_event: Optional[bool] = False,
+    ):
+        """
+        Runs an SQL statement, which can be data manipulation language (DML)
+        or data definition language (DDL)
+
+        :param cluster_identifier: unique identifier of a cluster
+        :type cluster_identifier: str
+        :param database: the name of the database
+        :type database: str
+        :param sql: the SQL statement text to run
+        :type sql: str
+        :param db_user: the database user name
+        :type db_user: str
+        :param parameters: the parameters for the SQL statement
+        :type parameters: list
+        :param secret_arn: the name or ARN of the secret that enables db access
+        :type secret_arn: str
+        :param statement_name: the name of the SQL statement
+        :type statement_name: str
+        :param with_event: indicates whether to send an event to EventBridge
+        :type with_event: bool
+
+        """
+        """only provide parameter argument if it is valid"""
+        if parameters:
+            response = self.get_conn().execute_statement(
+                ClusterIdentifier=cluster_identifier,
+                Database=database,
+                Sql=sql,
+                DbUser=db_user,
+                WithEvent=with_event,
+                SecretArn=secret_arn,
+                StatementName=statement_name,
+                Parameters=parameters,
+            )
+        else:
+            response = self.get_conn().execute_statement(
+                ClusterIdentifier=cluster_identifier,
+                Database=database,
+                Sql=sql,
+                DbUser=db_user,
+                WithEvent=with_event,
+                SecretArn=secret_arn,
+                StatementName=statement_name,
+            )
```

Comment:
```suggestion
        response = self.get_conn().execute_statement(
                ClusterIdentifier=cluster_identifier,
                Database=database,
                Sql=sql,
                DbUser=db_user,
                WithEvent=with_event,
                SecretArn=secret_arn,
                StatementName=statement_name,
                Parameters=parameters,
            )
```

Since `parameters` is not required https://docs.aws.amazon.com/redshift-data/latest/APIReference/API_ExecuteStatement.html

Reply:
boto3 behaved differently when passed "None" as Parameters vs not passing Parameters at all, hence the need fot the if-else